pytorch_lightning/plugins/training_type/single_tpu.py

```diff
@@ -54,13 +56,20 @@ def pre_dispatch(self) -> None:
         self.tpu_local_core_rank = xm.get_local_ordinal()
         self.tpu_global_core_rank = xm.get_ordinal()
 
-    def on_save(self, checkpoint: dict) -> dict:
```

Comment:
Do we not want to rely on the training type API?

Reply:
Wdym? `save_checkpoint` is part of the Accelerator API.

Reply:
I mean that maybe

```python
        if _OMEGACONF_AVAILABLE:
            checkpoint = apply_to_collection(checkpoint, (DictConfig, ListConfig), OmegaConf.to_container)
```

and the `callback` key filter

should be kept under `on_save`

and we just need a hook to override `atomic_save`

Reply:
Ahh, I see. So, `save_checkpoint` would have two hooks then (`on_save` and `save`)

Reply:
Yes, that's the idea.

Reply:
@carmocca This wouldn't work for `tpu_spawn`. In `save_checkpoint`, it only saves the checkpoint for global rank zero. But for TPUs, there's a rendevous in `xm.save` method hence all processes/ranks need to reach that part.